tensorflow/contrib/tensorrt/resources/trt_allocator.cc

```diff
@@ -37,8 +37,20 @@ void TRTCudaAllocator::free(void* memory) { cudaFree(memory); }
 
 void* TRTDeviceAllocator::allocate(uint64_t size, uint64_t alignment,
                                    uint32_t flags) {
+  // WAR for allocator alignment requirement. Certain cuda API calls require GPU
+  // memory with alignemtn to cudaDeviceProp::textureAlignment.
+  // See issue #20856
+  alignment = 512;
   assert((alignment & (alignment - 1)) == 0);  // zero or a power of 2.
-  void* mem = allocator_->AllocateRaw(alignment, size);
+  void* mem = allocator_->AllocateRaw(alignment, size + alignment);
+  CHECK(mem);
```

Comment:
Is it fine to leave nullptr for trt library to handle (and return error properly)?

Reply:
It should be fine according to their API: https://docs.nvidia.com/deeplearning/sdk/tensorrt-api/c_api/classnvinfer1_1_1_i_gpu_allocator.html
Will remove all CHECK and return nullptr instead.